### PR TITLE
shuffle and scramble

### PR DESCRIPTION
**Pull request by bgold-cosmos**
`shuffle` makes random permutations of Patterns, `scramble` randomly select elements from
Patterns.  Also includes `cycleChoose` as a helper function (which might be useful on its own).

### CODE REVIEW

File: Sound/Tidal/Pattern.hs [hunk 1]
```diff
@@ -1181,3 +1181,25 @@ swingBy::Time -> Time -> Pattern a -> Pattern a
...
+
+{- | `cycleChoose` is like `choose` but only picks a new item from the list
+once each cycle -}
+cycleChoose::[a] -> Pattern a
+cycleChoose xs = Pattern $ \(s,e) -> [((s,e),(s,e), xs!!(floor $ (dlen xs)*(ctrand s) ))]
+  where dlen xs = fromIntegral $ length xs
+        ctrand s = timeToRand $ fromIntegral $ floor $ sam s
+
+{- | `shuffle n p` divides one cycle of the `p` into `n` parts and returns a
+new random permutation each cycle.  For example, `shuffle 4 "bd sn cp hh"`
```
Comment (telephon) [inline Sound/Tidal/Pattern.hs]: what does "permutation" mean here? Permutation of n parts? The example doesn't really show how it works (or I just don't get it!).

Comment (bgold-cosmos) [inline Sound/Tidal/Pattern.hs]: Yes, basically it turns the pattern into a list of n parts, then rearranges them randomly back into a pattern.  I'll see if I can clarify the example a bit.

Comment (telephon) [inline Sound/Tidal/Pattern.hs]: ah so they are not the actual permutations, but random selections? So from a pattern `bd hh cp`` you could get the sequence `bd bd hh`, e.g. or are only `bd cp hh`, `cp bd hh` and `cp hh bd` possible?

Comment (bgold-cosmos) [inline Sound/Tidal/Pattern.hs]: No, they **are** permutations - that's the difference between `shuffle` (permutations) and `scramble` (random selections).  I think I'll just write out a lengthy example for clarity, it's not like space is at a premium.

Comment (bgold-cosmos) [inline Sound/Tidal/Pattern.hs]: Yes, maybe the new documentation is clearer?

File: Sound/Tidal/Pattern.hs [hunk 2]
```diff
@@ -1189,16 +1189,20 @@ cycleChoose xs = Pattern $ \(s,e) -> [((s,e),(s,e), xs!!(floor $ (dlen xs)*(ctra
   where dlen xs = fromIntegral $ length xs
         ctrand s = timeToRand $ fromIntegral $ floor $ sam s
 
-{- | `shuffle n p` divides one cycle of the `p` into `n` parts and returns a
-new random permutation each cycle.  For example, `shuffle 4 "bd sn cp hh"` 
-might return `"sn cp bd hh"` one cycle and `"hh cp bd sn"` the next.
+{- | `shuffle n p` evenly divides one cycle of the pattern `p` into `n` parts,
+and returns a random permutation of the parts each cycle.  For example,
```
Comment (telephon) [inline Sound/Tidal/Pattern.hs]: now it is very clear. Maybe "returns a randomly chosen permutation" ?

**Pull request merged**